### PR TITLE
fix some regressions in markdown highlight rules

### PR DESCRIPTION
**Pull request by kevinushey**
Closes https://github.com/rstudio/rstudio/issues/1699.

### CODE REVIEW

**Comment by kevinushey** [comment]
Note that there are still a number of issues / edge-cases in the Markdown parser -- for example:

    **Bold`Code ** Code`Bold**

should be highlighted as

**Bold`Code ** Code`Bold**

but that basically means we need to properly embed understanding of `` ` `` code blocks within the rules we use for finding `**`-delimited text. This is all definitely doable with Ace's highlight rules, but IMHO is beyond what we want to take on for v1.2.

File: src/gwt/acesupport/acemode/markdown_highlight_rules.js [hunk 1]
```diff
@@ -64,11 +64,10 @@ var ScalaHighlightRules = require("ace/mode/scala_highlight_rules").ScalaHighlig
 var ShHighlightRules = require("mode/sh_highlight_rules").ShHighlightRules;
 var StanHighlightRules = require("mode/stan_highlight_rules").StanHighlightRules;
 var SqlHighlightRules = require("mode/sql_highlight_rules").SqlHighlightRules;
-var JavaScriptHighlightRules = require("ace/mode/javascript_highlight_rules").JavaScriptHighlightRules;
```
Comment (jmcphers) [inline src/gwt/acesupport/acemode/markdown_highlight_rules.js]: Why was this removed? (unused var?)

Comment (kevinushey) [reply src/gwt/acesupport/acemode/markdown_highlight_rules.js]: It was accidentally duplicated - it's already defined a few lines up.
